src/backend/db/database_manager.py

```diff
@@ -259,3 +296,131 @@ def check_customer_guid_exists(self, customer_guid):
+            else:
+                logger.error(f"Failed to add column '{field_name}' in custom_field_values.")
+                session.rollback()
+                raise ValueError(f"Column '{field_name}' was not added successfully.")
+
+            session.commit()
+            return True
+        except IntegrityError as e:
+            logger.error(f"Duplicate field name error: {e}")
+            session.rollback()
+            raise ValueError("A custom field with this name already exists.")
+        except SQLAlchemyError as e:
+            logger.error(f"Error adding custom field: {e}")
+            session.rollback()
+            return False
+        finally:
+            session.close()
+
+    def list_custom_fields(self, customer_guid):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+        try:
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            query = '''SELECT * FROM custom_fields'''
+            results = session.execute(text(query)).fetchall()
+            logger.debug(results)
+
+            if results:
+                return [
+                    {"field_name": result[0] or "Unknown",  # Default if field_name is None
+                     "field_type": result[1] or "Unknown",  # Default if field_type is None
+                     "required": result[2] if result[2] is not None else False}  # Default if required is None
+                    for result in results
+                ]
+            else:
+                # If no results found, log and return an empty list
+                logger.info(f"No custom fields found for customer: {customer_guid}")
+                return []
+
+        except SQLAlchemyError as e:
+            # Log the error and return an empty list or raise an exception
+            logger.error(f"Error listing custom fields for customer {customer_guid}: {e}")
+            return []  # You could also raise a custom exception here, if needed
+        finally:
+            session.close()
+
+    def delete_custom_field(self, customer_guid, field_name):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+        try:
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the field exists
+            query_check = '''SELECT COUNT(*) FROM custom_fields WHERE field_name = :field_name'''
+            result = session.execute(text(query_check), {"field_name": field_name}).scalar()
+
+            if result == 0:
+                logger.debug(f"No custom field found with name: {field_name}")
+                return False  # Field does not exist
```

Comment:
You should return No custom field present instead of only returning false, so that the API will know why custom field deletion failed